src/WebJobs.Script.WebHost/Diagnostics/FunctionsVersionLoggerService.cs

```diff
@@ -0,0 +1,126 @@
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Script.Diagnostics
+{
+    /// <summary>
+    /// Service responsible for logging versions
+    /// </summary>
+    public class FunctionsVersionLoggerService : IHostedService, IDisposable
+    {
+        private const double IntervalInSecondsDev = 1;    // interval for dev unit testing (1 s) in which the logs would be dumped
+        private readonly ILogger<FunctionsVersionLoggerService> _logger;
+        private readonly Timer _timer;
+        private readonly TimeSpan _interval;
+        private readonly IEnvironment _environment;
+        private bool _disposed;
+        private double intervalInSecondsProd = 3600;      // default interval (1 hour) in which the logs would be dumped
+
+        public FunctionsVersionLoggerService(ILogger<FunctionsVersionLoggerService> logger, IEnvironment environment)
+        {
+            _timer = new Timer(OnTimer, null, Timeout.Infinite, Timeout.Infinite);
+            _environment = environment ?? throw new ArgumentNullException(nameof(environment));
+            _interval = TimeSpan.FromSeconds(this.Interval);
+            _logger = logger;
+        }
+
+        public virtual double Interval
+        {
+            get { return (_environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteSlotName) == "development") ? IntervalInSecondsDev : intervalInSecondsProd; }
+            set { intervalInSecondsProd = value; }
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            // start the timer by setting the due time
+            SetTimerInterval((int)_interval.TotalMilliseconds);
+
+            return Task.CompletedTask;
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            // stop the timer if it has been started
+            _timer?.Change(Timeout.Infinite, Timeout.Infinite);
+
+            return Task.CompletedTask;
+        }
+
+        private void OnTimer(object state)
+        {
+            PublishLogsSamplesAsync();
+        }
+
+        private void PublishLogs()
+        {
+            try
+            {
+                _logger.LogInformation("FunctionsExtensionVersion : {0}", _environment.GetEnvironmentVariable(EnvironmentSettingNames.FunctionsExtensionVersion));
+                _logger.LogInformation("Framework : {0}", _environment.GetEnvironmentVariable(EnvironmentSettingNames.Framework));
+                _logger.LogInformation("FrameworkVersion : {0}", _environment.GetEnvironmentVariable(EnvironmentSettingNames.FrameworkVersion));
+                _logger.LogInformation("SlotName : {0}", _environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteSlotName));
+            }
+            catch (Exception exc) when (!exc.IsFatal())
+            {
+                throw;
+            }
+        }
+
+        private void PublishLogsSamplesAsync()
```

Comment:
nit: remove `Async` from this method name....

Reply:
Done